source/Octopus.Server.Client/Exceptions/OctopusExceptionFactory.cs

```diff
@@ -51,74 +51,96 @@ public static async Task<OctopusException> CreateException(HttpResponseMessage r
-                    helpText = errors.HelpText;
-                    if (!string.IsNullOrWhiteSpace(errors.FullException))
-                    {
-                        fullDetails += Environment.NewLine + "Server exception: " + Environment.NewLine + errors.FullException + Environment.NewLine + "-----------------------" + Environment.NewLine;
-                    }
-                }
-            }
-            // ReSharper disable once EmptyGeneralCatchClause
-            catch
+                HelpText = errorsContract.HelpText
+            };
+        }
+
+        private static OctopusValidationException CreateOctopusValidationException(int statusCode, string body)
+        {
+            var errorsContract = OctopusErrorsContractFromBody(body);
+
+            return new OctopusValidationException(
+                statusCode,
+                errorsContract.ErrorMessage,
+                errorsContract.Errors,
+                errorsContract.Details)
+            {
+                HelpText = errorsContract.HelpText
+            };
+        }
+
+        private static OctopusServerException CreateOctopusServerException(int statusCode, string body)
+        {
+            var errorsContract = OctopusErrorsContractFromBody(body);
+            var fullDetails = errorsContract.ErrorMessage;
+            if (!string.IsNullOrWhiteSpace(errorsContract.FullException))
             {
+                fullDetails =
+                    $"Octopus Server returned an error: {errorsContract.ErrorMessage} {Environment.NewLine}Server exception: {Environment.NewLine}{errorsContract.FullException}{Environment.NewLine} ----------------------- {Environment.NewLine}";
             }
 
-            return new OctopusServerException(statusCode, fullDetails) {HelpText = helpText};
+            return new OctopusServerException(statusCode, fullDetails)
+            {
+                HelpText = errorsContract.HelpText
+            };
         }
 
-        static string GetErrorMessage(string body)
+        internal static OctopusErrorsContract OctopusErrorsContractFromBody(string body)
         {
+            OctopusErrorsContract result = null;
             string errorMessage;
+
             try
             {
-                var errors = JsonConvert.DeserializeObject<OctopusErrorsContract>(body);
-                errorMessage = errors.ErrorMessage;
+                result = JsonConvert.DeserializeObject<OctopusErrorsContract>(body);
+                errorMessage = result?.ErrorMessage ?? body;
+            }
+            catch (Exception jsonEx) when (jsonEx is ArgumentNullException or JsonSerializationException)
+            {
+                errorMessage = $"JSON deserialize error for {body} :: {jsonEx.Message}";
             }
-            catch
+            catch (Exception ex)
             {
-                errorMessage = body;
+                errorMessage = $"Unexpected json deserialization error for {body} :: {ex.Message}";
```

Comment:
These error messages are worded slightly differently, but effectively say the same thing. Why the different messages depending on exception type?

If it were me, it'd probably have left the behaviour of this alone, and just stuck the body in the error message. I'm not sure it's helpful to potentially give customers exception messages about failed deserialization because the Octopus payload isn't valid. This information is probably better in a log message.